### ISSUE TITLE
Add 'search page' function.

### ISSUE THREAD

**Issue by vext01** [jansc/ncgopher#5]
I couldn't find a way to search the current page, so I guess it isn't implemented yet? That'd be a nice addition.

We might want to re-think the key-bindings, as search is often (in stuff like `vim` and `less`) invoked with `/` or `?` (forward, backward) and then `n` to search again. Currently `n` highlights the next link in ncgopher.

Thanks!

**Comment by jansc** [comment]
You're right. Search is not implemented yet, but it's on the roadmap.

**Comment by SeriousBug** [comment]
Hey, I'd love to see this feature too! Are you accepting PRs? I'd be happy to give implementing this feature a try this weekend.

**Comment by jansc** [comment]
We'd be glad to receive a pull request! Sorry for the late answer :-(

**Comment by jansc** [comment]
I've now pushed a simple implementation of search in text. It works as follows:
- Press '/' to open search box and enter your query
- Pressing enter will search (so, no incremental search for now - even though it should be trivial to implement)
- Pressing escape or deleting all input will close the search box
- Once the search is executed, all occurrences of the search term will be highlighted and the current row is moved to the first occurrence of the search term. If no occurrences are below the current position, the first occurrence in the document will be highlighted instead (wrapping). If the search term is not found nothing happens (Should probably show a message)
- Press 'n' to move to the next search result
- Press 'N' to move to the previous search result
- Search is  case-sensitive (for now)

So the keyboard shortcuts are vim-like. As a consequence, I had to change 'n' which used to 'go to next link'. New shortcut for next/previous link is 'l' and 'L'. 'p' for previous link has been removed.
I also wanted to implement backward search, but the vim uses '?' for that and this key is taken for help. Any suggestions?
